crates/forge_analyzer/src/interp.rs

```diff
@@ -425,6 +425,25 @@ impl ValueManager {
                 .insert((def_id_func, var_id, projection_vec), value);
         }
     }
+
+    pub fn get_var_with_projection(
+        &self,
+        def_id_func: DefId,
+        var_id: VarId,
+        projection_vec: ProjectionVec,
+    ) -> Option<&Value> {
+        if let Some(value) =
+            self.varid_to_value_with_proj
+                .get(&(def_id_func, var_id, projection_vec.clone()))
```

Comment:
is there any way to not clone this here ...

Reply:
Unfortunately not, since `std`'s trait for heterogeneous lookups are kind of scuffed. If you want you can add the [hashbrown](https://github.com/rust-lang/hashbrown) crate(this is literally `std::collections::HashMap`, but with more features that they don't want to stabilize yet. You'd use their [get](https://docs.rs/hashbrown/latest/hashbrown/struct.HashMap.html#method.get) method which takes something that implements [Equivalent](https://docs.rs/hashbrown/latest/hashbrown/trait.Equivalent.html). So you'd do something like

```rust
#[derive(PartialEq, Eq, Clone, Copy, Hash)]
struct ProjLookup<'a>(DefId, VarId, &'a [Value]);

impl Equivalent<&(DefId, VarId, SmallVec<Value>)> for ProjLookup<'_> {
    fn equivalent(&self, key: &(DefId, VarId, SmallVec<Value>) -> bool {
        self.0 == key.0 && self.1 == key.1 && self.2 == key.2[..]
    }
}

if let Some(value) = self.varid_to_value_with_proj.get(&ProjLookup(def_id_func, var_id, &projection_vec[..])
```

Reply:
Will take a look at the hashbrown documentation. However, I think we should eventually refractor them all this way, so I will leave this out of this PR.